pkg/fleetautoscalers/controller.go

```diff
@@ -103,6 +111,13 @@ func NewController(
 		UpdateFunc: func(_, newObj interface{}) {
 			c.workerqueue.Enqueue(newObj)
 		},
+		DeleteFunc: func(fas interface{}) {
+			scaler, ok := fas.(*autoscalingv1.FleetAutoscaler)
+			if !ok {
+				return
+			}
+			c.removeFasThread(scaler)
+		},
```

Comment:
In `Run(ctx context.Context, workers int)` when the `ctx` returns a value from `Done()`, we should send a message to all the termination channels to make sure all the fasThreads are terminated.

Reply:
Good advice! Added it into `Run`.